src/main/kotlin/nl/info/zac/app/zaak/ZaakKoppelenRestService.kt

```diff
@@ -86,14 +79,60 @@ class ZaakKoppelenRestService @Inject constructor(
         addZoekVeld(ZoekVeld.ZAAK_IDENTIFICATIE, zoekZaakIdentifier)
         addFilter(FilterVeld.ZAAK_IDENTIFICATIE, FilterParameters(listOf(zaak.identificatie), true))
     }
-}
 
-private fun ZaakZoekObject.toRestZaakKoppelenZoekObject(linkable: Boolean) =
-    RestZaakKoppelenZoekObject(
-        id = identificatie,
-        type = ZoekObjectType.ZAAK,
-        identificatie = identificatie,
-        omschrijving = omschrijving,
-        statustypeOmschrijving = statustypeOmschrijving,
-        documentKoppelbaar = linkable,
-    )
+    private fun ZaakZoekObject.toRestZaakKoppelenZoekObject(linkable: Boolean) =
+        RestZaakKoppelenZoekObject(
+            id = getObjectId(),
+            type = ZoekObjectType.ZAAK,
+            identificatie = identificatie,
+            omschrijving = omschrijving,
+            statustypeOmschrijving = statustypeOmschrijving,
+            isKoppelbaar = linkable,
+        )
+
+    private fun filterSearchResults(
+        searchResults: ZoekResultaat<out ZoekObject>,
+        zaak: Zaak,
+        relationType: RelatieType
+    ): RestZoekResultaat<RestZaakKoppelenZoekObject> =
+        RestZoekResultaat(
+            searchResults.items.map {
+                val zaakZoekObject = it as ZaakZoekObject
+                zaakZoekObject.toRestZaakKoppelenZoekObject(
+                    isLinkable(sourceZaak = zaak, targetZaak = zaakZoekObject, relationType = relationType)
+                )
+            },
+            searchResults.count
+        )
+
+    private fun isLinkable(
+        sourceZaak: Zaak,
+        targetZaak: ZaakZoekObject,
+        relationType: RelatieType
+    ) = targetZaak.hasSameState(sourceZaak) &&
+        sourceZaak.hasLinkRights() &&
+        targetZaak.hasLinkRights() &&
+        targetZaak.isLinkableTo(sourceZaak, relationType)
+
+    private fun ZaakZoekObject.hasSameState(sourceZaak: Zaak) =
```

Comment:
I think `hasSameState` is quite unclear. The word `state` can mean many different things. I would make this function name much more explicit. It is about a zaak being open or not and about an archiefnominatie. I expect that back in the function name somehow. Otherwise it will surely confuse future developers